beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1653,30 +1667,49 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         }
     }
 
-    /// Fetch data_columns for a given block from the store.
+    /// Fetch all data_columns for a given block from the store.
     pub fn get_data_columns(
         &self,
         block_root: &Hash256,
-    ) -> Result<Option<DataColumnSidecarList<E>>, Error> {
+    ) -> Result<Vec<Arc<DataColumnSidecar<E>>>, Error> {
+        let mut items = vec![];
+        for entry_result in self.blobs_db.iter_column_from::<Vec<u8>>(
+            DBColumn::BeaconDataColumn,
+            &get_data_column_key(block_root, &0),
+            &get_data_column_key(block_root, &u64::max_value()),
```

Comment:
why `u64::max_value` here rather than `self.spec.number_of_columns`?

Reply:
Both should be equal. Using u64 max value is more expressive to requesting the entire column key space for the prefix `block_root`. This code is asking: give me all entries that start with `block_root` regardless of their value.

Reply:
Removed in https://github.com/sigp/lighthouse/pull/5890/commits/7d72c364eb7142eef1ca20b3f5fe4d261bf67d2e